hack/build.sh.d/func.sh

```diff
@@ -0,0 +1,18 @@
+#!/usr/bin/env bash
+
+# There no midstream image currently, fallback to upstream one.
+readonly func_util_image="${KN_PLUGIN_FUNC_UTIL_IMAGE:-quay.io/boson/alpine-socat:1.7.4.3-r1-non-root}"
+readonly func_repo_ref="${FUNC_REPO_REF:-openshift-knative/kn-plugin-func}"
+readonly func_repo_branch_ref="${FUNC_REPO_BRANCH_REF:-serverless-1.31}"
+readonly default_node_builder_image="${DEFAULT_NODE_BUILDER_IMAGE:-registry.access.redhat.com/ubi8/nodejs-16}"
+readonly default_quarkus_builder_image="${DEFAULT_QUARKUS_BUILDER_IMAGE:-registry.access.redhat.com/ubi8/openjdk-17}"
+readonly default_python_builder_image="${DEFAULT_PYTHON_BUILDER_IMAGE:-registry.access.redhat.com/ubi8/python-39}"
+
+export EXTERNAL_LD_FLAGS="${EXTERNAL_LD_FLAGS:-} \
+-X knative.dev/func/pkg/k8s.SocatImage=${func_util_image} \
+-X knative.dev/func/pkg/k8s.TarImage=${func_util_image} \
+-X knative.dev/func/pkg/pipelines/tekton.FuncRepoRef=${func_repo_ref} \
+-X knative.dev/func/pkg/pipelines/tekton.FuncRepoBranchRef=${func_repo_branch_ref} \
```

Comment:
@matejvasek @jrangelramos @Kaustubh-pande FYI

I've added those two new hooks. In case we need to override these values and for the future reuse as well. So we don't forget about it in later releases. Pls double check names, values for me, thanks!

Reply:
Looking at the env var names, we should probably rename all of them to `FUNC_` suffix. But that's for another day.